### PR TITLE
Switch to bootstrap icons

### PR DESCRIPTION
**Pull request by surayya-MS**
# Switch to bootstrap icons

Part of https://github.com/dotnet/aspnetcore/issues/48341
Before:
open-iconic-before
After:
after

### CODE REVIEW

**Review by javiercn — APPROVED** [review]

**Review by Eilon — CHANGES_REQUESTED** [review]
Also wondering in general if there are any notable visual differences between the old and new icons?

**Comment by SteveSandersonMS** [comment]
> Also wondering in general if there are any notable visual differences between the old and new icons?

I assume the new icons are not identical but from the naming it sounds like @surayya-MS has chosen ones similar to the old ones, which sounds ideal to me. I'd be interested to see a screenshot of the result though!

**Comment by surayya-MS** [comment]
@SteveSandersonMS , @Eilon I attached the screenshots to the description

**Comment by SteveSandersonMS** [comment]
Looks great! Definitely better than the old icons.

File: src/Components/Samples/BlazorServerApp/Pages/_Layout.cshtml [hunk 1]
```diff
@@ -10,6 +10,7 @@
     <base href="~/" />
     <link rel="stylesheet" href="css/bootstrap/bootstrap.min.css" />
     <link href="css/site.css" rel="stylesheet" />
+    <link rel="stylesheet" href="css/bootstrap-icons/bootstrap-icons.css" />
```
[the remark this answers is not in the document]
Comment (surayya-MS) [reply src/Components/Samples/BlazorServerApp/Pages/_Layout.cshtml]: Done

**Comment by Eilon** [comment]
> @SteveSandersonMS , @Eilon I attached the screenshots to the description. I tried to choose similar ones https://icons.getbootstrap.com/

Yeah looks good! I wasn't expecting identical, I just wasn't sure if they were conceptually different. Nice!

**Review by Eilon — DISMISSED** [review]
Got another fun one: I think we need to update the appropriate Third Party Notices (3PN) text files to reflect this change as well. The notices file lists the various projects and licenses of other things that we either include in this repo, or re-ship to users (e.g. stuff that's in templates).

1. The Bootstrap stuff that we currently list might be out of date (?)
2. The Bootstrap Icons stuff seems like it's a separate project, and we would need to add it.

There are two 3PN files that I think might need to be updated:
1. https://github.com/dotnet/aspnetcore/blob/main/THIRD-PARTY-NOTICES.txt --> this should list every project/license that is checked into this repo, regardless of why we include it
2. https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/THIRD-PARTY-NOTICES --> this should list every project/license that is part of the templates that we ship to users

Having said all of this, it is probably best to confirm with this repo's owners to verify what process should be used for this.

**Comment by surayya-MS** [comment]
@Eilon I'm not sure where I could get the license for bootstrap icons and where to include it. @SteveSandersonMS , @javiercn , @danroth27 do you know?

My findings so far:

- open iconic wasn't included in these 2 files https://github.com/dotnet/aspnetcore/blob/main/THIRD-PARTY-NOTICES.txt and https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/THIRD-PARTY-NOTICES
- open iconic however has this license inside its folder https://github.com/dotnet/aspnetcore/blob/6575b289081b9f85fb30d01c7b306d1ec5d00a68/src/Components/Samples/BlazorServerApp/wwwroot/css/open-iconic/ICON-LICENSE
- Bootstrap license is included in one of the mentioned files https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/THIRD-PARTY-NOTICES
- There is a license file inside `Source code zip`. It can be downloaded [here](https://github.com/twbs/icons/releases/tag/v1.10.5). But I'm not sure if this is the right one for bootstrap icons because the icons are in the bootstrap-icons-1.10.5.zip and there wasn't any license file, only some comments about license and copyright in bootstrap-icons.css and min.css (see in this PR)
- https://getbootstrap.com/docs/4.0/about/license/ here it says "Keep the license and copyright notice included in Bootstrap’s CSS and JavaScript files when you use them in your works"

**Comment by Eilon** [comment]
> @javiercn, @Eilon should I also switch to bootstrap icons in maui blazor template?

Yeah perhaps log a bug in dotnet/maui for now, and link to the AspNetCore issue/PR.

**Comment by Eilon** [comment]
> @Eilon I'm not sure where I could get the license for bootstrap icons and where to include it. @SteveSandersonMS , @javiercn , @danroth27, @mkArtakMSFT do you know?
> 
> My findings so far:
> 
> * open iconic wasn't included in these 2 files https://github.com/dotnet/aspnetcore/blob/main/THIRD-PARTY-NOTICES.txt and https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/THIRD-PARTY-NOTICES

It probably _should_ have been, but too late now.

> * open iconic however has this license inside its folder https://github.com/dotnet/aspnetcore/blob/6575b289081b9f85fb30d01c7b306d1ec5d00a68/src/Components/Samples/BlazorServerApp/wwwroot/css/open-iconic/ICON-LICENSE

Yeah we preserve the OSS stuff as-is, including any license files they already have.

> * Bootstrap license is included in one of the mentioned files https://github.com/dotnet/aspnetcore/blob/main/src/ProjectTemplates/THIRD-PARTY-NOTICES

That's good. Does it match the version that we use?

> * There is a license file inside `Source code zip`. It can be downloaded [here](https://github.com/twbs/icons/releases/tag/v1.10.5). But I'm not sure if this is the right one for bootstrap icons because the icons are in the bootstrap-icons-1.10.5.zip and there wasn't any license file, only some comments about license and copyright in bootstrap-icons.css and min.css (see in this PR)
> * https://getbootstrap.com/docs/4.0/about/license/ here it says "Keep the license and copyright notice included in Bootstrap’s CSS and JavaScript files when you use them in your works"

The `wwwroot/css/bootstrap-icons/bootstrap-icons[.min].css` files in this PR say `Licensed under MIT (https://github.com/twbs/icons/blob/main/LICENSE)`, so I would personally go as far as to mention that we use `Bootstrap Icons v1.10.5` and copy the license text of https://github.com/twbs/icons/blob/main/LICENSE as the entry for it in the 3rd party notices. In my view it's two projects (Bootstrap, and Bootstrap Icons), that happen to use the same license.

But........ please confirm with someone who has more authority in this repo 😄 Back in my day in this repo, that's what I would have done.

**Comment by surayya-MS** [comment]
Thanks @Eilon!

I added the license for Bootstrap icons.

Bootstrap was updated to 5.1.0 https://github.com/dotnet/aspnetcore/pull/35197. I now updated the 3PN too

**Review by SteveSandersonMS — APPROVED** [review]
Superb! This looks ideal.

**Comment by mkArtakMSFT** [comment]
I'm dismissing @Eilon's "requested changes" now given he is away. Looks like @surayya-MS has done all that is needed for now!

**Pull request merged**